lib/mobilecommons.js

```diff
@@ -18,71 +17,63 @@ const auth = {
 };
 
 /**
-* Mobile Commons profile_update API. Can be used to subscribe the user to an
-* opt-in path.
-*
+* Mobile Commons profile_update API. Can be used to subscribe the user to an opt-in path.
 * @see https://mobilecommons.zendesk.com/hc/en-us/articles/202052534-REST-API#ProfileUpdate
 *
-* @param phone
-*   Phone number of the profile to update.
-* @param optInPathId
-*   Opt-in path to subscribe the user to.
-* @param customFields
-*   Object with MoCo custom profile field names as properties, and values to update the user with.
-*   Note: Field names are case-sensitive.
+* @param {object} profileId - Mobile Commons Profile ID
+* @param {number} phone - Phone to opt in
+* @param {string} oip - Opt-in Path to post to
+* @param {object} updateFields - field values to update on current User's Mobile Commons Profile
 */
-
-exports.profile_update = function(phone, optInPathId, customFields) {
+exports.profile_update = function(profileId, phone, oip, updateFields) {
+  const statName = 'mobilecommons: POST profile_update';
   if (process.env.MOBILECOMMONS_DISABLED) {
     logger.warn('MOBILECOMMONS_DISABLED');
 
     return;
   }
 
-  logger.log('debug', 'mobilecommons.profile_update for mobile:%s oip:%s customFields:%s', phone, optInPathId, JSON.stringify(customFields));
-
-  var url = 'https://secure.mcommons.com/api/profile_update';
-  var authEmail = process.env.MOBILECOMMONS_AUTH_EMAIL;
-  var authPass = process.env.MOBILECOMMONS_AUTH_PASS;
+  const updateTxt = JSON.stringify(updateFields);
+  // Use profileId if available to avoid logging.
+  let identifier = `profile:${profileId}`;
+  if (!profileId) {
+    identifier = `phone:${phone}`;
```

Comment:
But you already set it  it 2 lines above to `profile:${profileId}`. Would it make better sense to first check `if (!profileId)`?